app/src/sharedTest/java/org/oppia/android/app/policies/PoliciesFragmentTest.kt

```diff
@@ -246,6 +272,28 @@ class PoliciesFragmentTest {
     }
   }
 
+  @Test
+  fun testPoliciesFragment_inTermsOfServicePage_clickOnPrivacyLink_opensPrivacyPolicyPage() {
+    launch<PoliciesFragmentTestActivity>(
+      createPoliciesFragmentTestActivity(
+        getApplicationContext(),
+        PolicyPage.TERMS_OF_SERVICE
+      )
+    ).use { scenario ->
+      scenario.onActivity {
+        it.mockCallbackListener = mockRouteToPoliciesListener
+        onView(withId(R.id.policy_description_text_view))
+          .check(matches(isDisplayed()))
+        onView(withId(R.id.policy_description_text_view))
+          .check(matches(withText(containsString("Privacy Policy"))))
+          .perform(click())
```

Comment:
Is this actually performing the click? I wonder if we still need to retrieve the spans from the text (not parsing, but just retrieving the ones that should already be there) in order to perform the click.

Reply:
No this was the issue. I added clickable spans and now the mock works.